### PR TITLE
feat(benchmark): display missing requests

### PR DESCRIPTION
**Pull request by snutij**
### Motivation

Closes https://github.com/Shopify/ruby-lsp/issues/504

Display in benchmarks results if there is any missing request benchmark. 

### Implementation

The second option proposed here: https://github.com/Shopify/ruby-lsp/issues/504#issuecomment-1499068788

* Map all classes which inherit from `BaseRequest` or `Listener`
* Require everything under `ruby_lsp/requests`
* Re-map all classes which inherit from `BaseRequest` or `Listener`
* If a delta is present, display it. 

Two open questions: 

* Is the output OK this way? Should I format it like removing `RubyLsp::Requests` (or any other way)?

```bash
================================================================================
Missing benchmarks:

RubyLsp::Requests::PathCompletion
RubyLsp::Requests::CodeLens
RubyLsp::Requests::CodeActionResolve
```

* Should I edit `github/workflows/benchmark.yml` to make it fail if there are any missing requests?

### Automated Tests

I think there is no test on the task, but I may be wrong. Feel free to point it out if necessary.

### Manual Tests

There are already missing benchmarks so: 

* run `bin/benchmark` 
* output `PathCompletion`, `CodeLens` & `CodeActionResolve`
* comment an item in `requests` hash, re-run `bin/benchmark` it should be added on missing benchmarks

### CODE REVIEW

**Comment by snutij** [comment]
I've signed the CLA!

**Comment by vinistock** [comment]
The implementation looks great! The only issue is we now need to add the missing benchmarks for CI to pass 😂. Would you like to take a stab at that too? Otherwise, we'll need to add the missing ones before we can merge this PR.

**Comment by snutij** [comment]
I tried to add the missing benchmarks, but I'm not sure I did it right. 😆 Feel free to let me know if I should do something different!

File: bin/benchmark [hunk 1]
```diff
@@ -298,4 +301,19 @@ if File.exist?(CACHE_FILE_PATH)
...
+
+all_requests = RubyLsp::Requests::BaseRequest.subclasses.concat(RubyLsp::Listener.subclasses)
+
+missing_benchmarks = all_requests - requests_benchmarked
+
+if missing_benchmarks.any?
+  puts "\n\n"
+  puts "=" * 80
+  puts "Missing benchmarks:\n\n"
+  puts missing_benchmarks.map(&:to_s)
```
Comment (vinistock) [inline bin/benchmark:316]: Here we need to `exit(1)`

**Review by vinistock — APPROVED** [review]

**Comment by vinistock** [comment]
Thank you for the contribution!

**Pull request merged**